### PR TITLE
Update 'spdlog' to v1.2.1

### PR DESCRIPTION
**Pull request by fantasticfears**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/update.html)
  step by step carefully. **Yes**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/fantasticfears/hunter/builds/19671025
  * https://travis-ci.com/fantasticfears/hunter/builds/88652199

---

Pending test suite PR https://github.com/ingenue/hunter/pull/280

* I've checked this [Git style guide](https://0.readthedocs.io/en/latest/git.html). **Yes**
* I've checked this [CMake style guide](https://0.readthedocs.io/en/latest/cmake.html). **Yes**
* My change will work with CMake 3.2 (minimum requirement for Hunter). **Yes**
* I will try to keep this pull request as small as possible and will try not to mix unrelated features. **Yes**

---

### CODE REVIEW

File: cmake/projects/spdlog/hunter.cmake [hunk 1]
```diff
@@ -73,7 +73,18 @@ hunter_add_version(
...
-)  
+)
+
+hunter_add_version(
+    PACKAGE_NAME
+    spdlog
+    VERSION
+    "1.2.1"
+    URL
+    "https://github.com/gabime/spdlog/archive/v1.2.1.tar.gz"
```
Comment (ruslo) [inline cmake/projects/spdlog/hunter.cmake]: There are some patches needed to adapt upstream version for Hunter, e.g. use Hunter to install "fmt" library: https://github.com/hunter-packages/spdlog/blob/hunter-0.16.3/CMakeLists.txt#L49-L52

I've created branch `hunter-1.2.1` based on upstream 1.2.1 tag:
* https://github.com/hunter-packages/spdlog/tree/hunter-1.2.1

Please send there pull request with patches from previous `hunter-0.6.13` branch:
* https://github.com/hunter-packages/spdlog/tree/hunter-0.16.3

Comment (fantasticfears) [reply cmake/projects/spdlog/hunter.cmake]: The upstream bundled `fmt` library. https://github.com/gabime/spdlog/tree/v1.2.1/include/spdlog/fmt Hunterized package is not needed anymore.

Comment (ruslo) [reply cmake/projects/spdlog/hunter.cmake]: Package manager should be responsible for managing dependencies, not package itself. In general the usage of bundled dependencies is a dirty hack, not a solution.

* https://docs.hunter.sh/en/latest/faq/why-do-we-need-forks.html#solution-for-bundled-sources
* https://cgold.readthedocs.io/en/latest/tutorials/install/managing-dependencies.html
* https://docs.hunter.sh/en/latest/creating-new/create/cmake-dependencies.html
* https://wiki.gentoo.org/wiki/Why_not_bundle_dependencies

Comment (fantasticfears) [reply cmake/projects/spdlog/hunter.cmake]: It makes sense. https://github.com/hunter-packages/spdlog/pull/7. I can help package it and update this PR with new hunterized spdlog.

**Comment by fantasticfears** [comment]
I think we missed something in packing `spdlog` https://travis-ci.com/fantasticfears/hunter/jobs/154375205 fmt::fmt is linked following INTERFACE.
Line 1692 says hunter found fmt package. Line 2287 says failure of finding fmt package

**Comment by ruslo** [comment]
> /usr/local/clang-5.0.0/bin/clang++ -DSPDLOG_FMT_EXTERNAL=1 ... -isystem /.../hunter/_testing/Hunter/_Base/e94d768/8fed42d/8d8dca6/Install/include ... -c /.../spdlog/Source/tests/test_fmt_helper.cpp

> /.../spdlog/Source/include/spdlog/fmt/fmt.h:23:10: fatal error: 'fmt/core.h' file not found

Path to includes is there, so I don't know what is the problem. Can you test it locally and check if `fmt/core.h` is installed?

**Comment by fantasticfears** [comment]
Ok, it's not there. This is clear to me now. fmt 5 upgraded the headers. Anyway to require fmt 5 in hunter? I'm going to file a PR to upgrade fmt firstly. Then we can come back to this.

**Comment by ruslo** [comment]
> I'm going to file a PR to upgrade fmt firstly.

Okay, thanks. `fmt` package sources used from upstream, should be straightforward. I've updated toolchains in `pkg.fmt` branch also.

**Comment by fantasticfears** [comment]
New tests are all green now. Thank you for the help.
https://travis-ci.com/fantasticfears/hunter/builds/89415943
https://ci.appveyor.com/project/fantasticfears/hunter/builds/19855062

**Comment by fantasticfears** [comment]
I messed up the pr branch. Here is the tests:

- https://ci.appveyor.com/project/fantasticfears/hunter/builds/19876092
- https://travis-ci.com/fantasticfears/hunter/builds/89487292

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/builds/19876820
* https://travis-ci.org/ingenue/hunter/builds/447749526?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/builds/19877220
* https://travis-ci.org/ingenue/hunter/builds/447755190?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.39

**Pull request merged**